### ISSUE TITLE
shapr3d  5.381.5210.0

### ISSUE THREAD

**Issue by DNS3K** [sl4v3k/Shapr3d_crack#5]
the newest [shapr3d.dll](https://mega.nz/file/OqRCBKzC#4bl4iezJ3WbKnaXyFLpeF5TcoIJsGgcsbl31nH_EPMw) might i get some help with this one? i cant seem to narrow the "call" down. i can modify the dll.

it comes from the non updating standalone ver from the website not the store but still installs as a windows app.

thanks 
xdns3kx
at gmail

**Comment by sl4v3k** [comment]
try this: repower.cc/Shapr3D.7z and let me know. This dll looks different than from app store.

**Comment by DNS3K** [comment]
From the msix package doesnt auto update
https://download.shapr3d.com/windows/latest
DOWNLOAD from shapr3d

Will try after work tonight

Thanks for the help

**Comment by sl4v3k** [comment]
tested, all works!! here is the place in dll from the link You sent:
.text:0000000181910FBF C6 45 01 00                             mov     [rbp+60h+var_5F], 0
.text:0000000181910FC3 48 8D 0D 16 43 BD FE                    lea     rcx, off_1804E52E0
.text:0000000181910FCA 4C 8D 45 01                             lea     r8, [rbp+60h+var_5F]
**.text:0000000181910FCE 48 8B C1                                mov     rax, rcx
.text:0000000181910FD1 90                                      nop
.text:0000000181910FD2 90                                      nop**
.text:0000000181910FD3 48 8B D8                                mov     rbx, rax
.text:0000000181910FD6 48 8D 0D F3 55 72 FE                    lea     rcx, unk_1800365D0
.text:0000000181910FDD FF 15 DD FE 35 FF                       call    cs:RhpNewFast
.text:0000000181910FE3 48 8B F8                                mov     rdi, rax
.text:0000000181910FE6 48 8B C8                                mov     rcx, rax
.text:0000000181910FE9 FF 15 41 7A 35 FF                       call    qword ptr cs:byte_180C66001+2A2Fh
.text:0000000181910FEF 48 8B 56 20                             mov     rdx, [rsi+20h]

**Comment by sl4v3k** [comment]
just test dll i gave You before, must work, let me know the results

**Comment by DNS3K** [comment]
Yes, it worked. Had some permissions issues with the directory but it's
fixed now. Thank you for your help

**Comment by r3041436** [comment]
Help, I can't find E8 4D 2E 80 FF. I download shapr3d 5.381.5210.0
and I download HxD to edit, and I failed.
This is my shapr3d.dll in driver link, help me and teach me, please.
https://drive.google.com/file/d/1ewAALZ2hotiVC6ipwofkuznNL47_YJhK/view?usp=sharing

thanks a lot

**Comment by h0kay** [comment]
Really appreciate your continued effort on this. Any chance you can post the call location on the latest DLL version (5.440)?

I have the DLL uploaded here:
https://drive.google.com/file/d/11aeQK7Ta8cNQseNi7crt-ECQUAFfVdU3/view?usp=sharing

The download path to the newest version in case it's needed is here:
download.shapr3d.com/windows/Shapr3D.Package_5.440.5599.0_Test/Shapr3D.Package_5.440.5599.0_x64_ARM64.msixbundle

I downloaded IDA to take a look at the code but didn't have much luck.

Thank you!

**Comment by sl4v3k** [comment]
check github

------- Original Message -------

**Comment by leeskype** [comment]
What's your modification logic? Can I study? My shapr3D version is 5.510.6062.0_x64. I tried to decompile its .dll file with IDA but I can't find the entry point, can you teach me if you can? Or can you help me modify my .dll file, the onedrive cloud disk link is as follows, https://hhu2016-my.sharepoint.com/:u:/g/personal/hhuyangli_hhu_edu_cn/EdNv_CR2oj9AluxClzvpfjIBVAxOz4J4Xp0-LRTiJOEnrg?e=eOV9bB

**Comment by leeskype** [comment]
thx

**Comment by sl4v3k** [comment]
Check updates, in cass of 5.510.6062.0_x64 file offset is under 0x1B47489 (in case You use hex editor)
in IDA You can see it under: 0x181B49289
.text:0000000181B49289 E8 22 FD 78 FF   call    lic_call_1812D8FB0 <--------- mov rax,rcx (48 8B C1 90 90)

**Comment by leeskype** [comment]
> 检查更新，如果5.510.6062.0_x64，偏移量低于 IDA 中的 0x1B47489（如果您使用十六进制编辑器） 您可以在以下位置看到它： 0x181B49289 .text：0000000181B49289 E8 22 FD 78 FF 调用lic_call_1812D8FB0 <--------- mov rax，rcx （48 8B C1 90 90）

It worked, thank you!

**Comment by leeskype** [comment]
> Check updates, in case of 5.510.6062.0_x64, offset is under 0x1B47489 (in case You use hex editor)
> in IDA You can see it under: 0x181B49289
> .text:0000000181B49289 E8 22 FD 78 FF call lic_call_1812D8FB0 <--------- mov rax,rcx (48 8B C1 90 90)

It worked, thank you!

**Comment by r3041436** [comment]
Thanks bro
I’ll check and learn

leeskype ***@***.***> 於 2023年11月23日 星期四寫道：

**Comment by Dmitriy1212121** [comment]
hello dear sl4m3l, can you help me with the Shape3d v5.381.5210.0 program? I can't find the line to change, I tried your patches did not help, thank you very much for your help, I will be very grateful for your help, here is the link to the file https://drive.google.com/file/d/1XfiLlBD_nNabGFcSO6Ee7mxAH0s1LzUQ/view?usp=sharing

**Comment by sl4v3k** [comment]
destription is clear, just search over binaries for patern from description, check also inmemory ptcher, all is there

**Comment by Dmitriy1212121** [comment]
thank you very much now I know where to look for your help is very valuable to me

**Comment by r3041436** [comment]
Hello there! After a year, I’m still struggling, so could you please provide a more detailed tutorial? Although I’m not currently using “Shapr3D,” I’m eager to learn this skill—and perhaps even more. I hope you won’t hesitate to guide me so that I can master it. As for the “metode Macajewa,” I tried Googling it, but I couldn’t find the key points. I’d greatly appreciate it if you could spare some time to be my teacher. 🙏

**Comment by DNS3K** [comment]
Hey guys as far as a hex editor that's pretty simple to even dumb luck your
way through it use 010 editor it's what allowed me to find what I was
looking for using just the dll and the instructions.

**Comment by h0kay** [comment]
Honestly I was clueless on how to do any of this too, I used chatgpt to help me and got it figured out pretty quickly afterwards.

**Comment by sl4v3k** [comment]
Hey, Shapr3d is quite easy target, no binary protection (no packer, no virtualization, no dynamic code encryption, no atidebug) all is about attaching debugger (x64dbg.com) to the shapr process and going over code and see what is what. Not really sure if I can teach You @r3041436 something more. Of courese first of all it is necessary to know Windows internals to be able to understand how app is working, understand intel opcodes etc.

**Comment by r3041436** [comment]
Thank you all for your careful responses. I will try following the steps again and test with a different version of the "dll" in Shapr3d. If I remember correctly, I made a mistake last year in the offset part. I used the offset provided by DNS3K, but couldn't find the correct value. I'm also curious, if no one prompts me with the offset today, what keywords should I use to search for where to modify the value?

**Comment by sl4v3k** [comment]
Please make analysis of cpp file of one of the inmemory patcher project, lets take latest one
`Shapr3D-5.590.6756.0 #60049c57--inmemory-patcher/Shapr3D-inmemory-patcher-2/Shapr3D-inmemory-patcher-2.cpp`
where You have place where 5 bytes of patch are injectec into shapr process space, there You can find exact offset:
```
        if (!WriteProcessMemory(shaprh, (LPVOID)(shapr3d_dll_base_addr + 0x1C1A436 + 0x1000), (LPCVOID)ptrAddress, 5, &bytes_written))
        {
            printf("WriteProcessMemory failed! error code -> 0x%08x\n", GetLastError());
        }
```
what in this case is `shapr3d_dll_base_addr + 0x1C1A436 + 0x1000`
if We assume shapr3d_dll_base_addr is 0 so offset is 0x1c1b436 (in memory)
in dll file should be 0x0x1C1A436
if dll is loaded to IDA than offset is taken from PE header and by defult in this case is 0x0180001000

**Comment by sl4v3k** [comment]
Actual keyword is changing from version to version so it is not possible to give on generic, You have to look manualy for similarities aroud interested function
